modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1665,12 +1695,12 @@ private CompletableFuture<Object> processMultiEntryAction(ReadWriteMultiRowRepli
                     }
 
                     if (rowIdsToDelete.isEmpty()) {
-                        return completedFuture(result);
+                        return completedFuture(didNoWrites(result));
                     }
 
-                    return updateAllCommand(request, rowIdsToDelete, txCoordinatorId)
+                    return chooseOpTsValidateAndBuildUpdateAllCommand(request, rowIdsToDelete, txCoordinatorId)
```

Comment:
It seems that we must choose the timestamp after we take locks (otherwise an operation that already took a timestamp [and hence must be executed before operations that take ts after it] and cannot take a lock for a long time would block all other RW operations on the same partition), so we cannot do it before this method.